Sources/TuistKit/Commands/TrackableCommand/CommandEventFactory.swift

```diff
@@ -8,15 +8,26 @@ import TuistSupport
 /// from different sources and tells `analyticsTagger` to send the event to a provider
 
 public final class CommandEventFactory {
+    private let environment: Environmenting
     private let machineEnvironment: MachineEnvironmentRetrieving
+    private let gitHandler: GitHandling
+    private let gitRefReader: GitRefReading
 
     public init(
-        machineEnvironment: MachineEnvironmentRetrieving = MachineEnvironment.shared
+        environment: Environmenting = Environment.shared,
+        machineEnvironment: MachineEnvironmentRetrieving = MachineEnvironment.shared,
+        gitHandler: GitHandling = GitHandler(),
+        gitRefReader: GitRefReading = GitRefReader()
     ) {
+        self.environment = environment
         self.machineEnvironment = machineEnvironment
+        self.gitHandler = gitHandler
+        self.gitRefReader = gitRefReader
     }
 
-    public func make(from info: TrackableCommandInfo) -> CommandEvent {
+    public func make(from info: TrackableCommandInfo) throws -> CommandEvent {
+        let commitSHA = try? gitHandler.currentCommitSHA()
```

Comment:
I added a `isInGitRepository` command. The best way to assert if we're in a git repository is through `git rev-parse` which returns a non-zero exit code when we're _not_ in a git repository. Since we need `git` to be able to run that, that command will also return `false` if `git` is not available.

We could add another check for `gitAvailable` but it won't change the behavior at all, so I'd consider it pointless for now.